core/src/tx.rs

```diff
@@ -45,6 +45,34 @@ pub enum AcceptTransactionFail {
     SignatureVerification(#[source] SignatureVerificationFail<TransactionPayload>),
 }
 
+fn instruction_size(isi: &InstructionBox) -> usize {
```

Comment:
On one hand I feel that it would be a more composable way to re-write it as a trait (`impl InstructionSize for InstructionBox`, `impl InstructionSize for RegisterBox` etc). On the other hand, this function looks quite minimal and just works, why not.